compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1037,6 +1041,19 @@ private boolean isTupleTypeAssignable(BType source, BType target, Set<TypePair>
             unresolvedTypes.add(pair);
         }
 
+        if (target.tag == TypeTags.JSON && source.tag == TypeTags.TUPLE) {
+            BTupleType rhsTupleType = (BTupleType) source;
+            for (int i = 0; i < rhsTupleType.tupleTypes.size(); i++) {
+                if (!isAssignable(rhsTupleType.tupleTypes.get(i), target, unresolvedTypes)) {
```

Comment:
Shall we extract `rhsTupleType.tupleTypes` out to a variable? Or simply use a for loop.

```suggestion
            for (BType tupleType : rhsTupleType.tupleTypes) {
                if (!isAssignable(tupleType, target, unresolvedTypes)) {
```